runtime/v2/runc/pause/sandbox.go

```diff
@@ -0,0 +1,104 @@
+//go:build linux
+// +build linux
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pause
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/pkg/shutdown"
+	"github.com/containerd/ttrpc"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/containerd/containerd/plugin"
+	api "github.com/containerd/containerd/runtime/v2/task"
+)
+
+func init() {
+	plugin.Register(&plugin.Registration{
+		Type: plugin.TTRPCPlugin,
+		ID:   "pause",
+		Requires: []plugin.Type{
+			plugin.InternalPlugin,
+		},
+		InitFn: func(ic *plugin.InitContext) (interface{}, error) {
+			ss, err := ic.GetByID(plugin.InternalPlugin, "shutdown")
+			if err != nil {
+				return nil, err
+			}
+
+			return &pauseService{
+				shutdown: ss.(shutdown.Service),
+			}, nil
+		},
+	})
+}
+
+// pauseService is an extension for task v2 runtime to support Pod "pause" containers via sandbox API.
+type pauseService struct {
+	shutdown shutdown.Service
+}
+
+var _ api.SandboxService = (*pauseService)(nil)
+
+func (p *pauseService) RegisterTTRPC(server *ttrpc.Server) error {
+	api.RegisterSandboxService(server, p)
+	return nil
+}
+
+func (p *pauseService) StartSandbox(ctx context.Context, req *api.StartSandboxRequest) (*api.StartSandboxResponse, error) {
```

Comment:
Confused, is this expected to startup a pause container here or is the client supposed to create the pause container with this sandbox id?

Reply:
My understanding is this is just a placeholder implementation so that the interface can be exercised and the pause container will be created by the client in this sandbox. The implementation for pause container replacement can go here by either starting a container or just implementing the same pause logic.